gapic/templates/%namespace/%name_%version/%sub/services/%service/_client_macros.j2

```diff
@@ -265,3 +267,26 @@
 
 {% macro define_extended_operation_subclass(extended_operation) %}
 {% endmacro %}
+
+{% macro auto_populate_uuid4_fields(api, method) %}
```

Comment:
The ads team is moving away from custom templates. Having common macros between the ads and standard templates would result in additional clean up once we remove the ads templates.

Reply:
But until they do move away, it's duplicate code. If we can have ads import a macro file that lives in the standard template directory (or do we need a symlink?) then when ads removes their templates, there will be nothing to do. At worst we wind up with two macro files: one that was intended only for standard templates, one that was intended for both templates.

I think this would be more elegant, particularly if the Ads migration takes a while. But I won't belabor the point or block on this; it could hurt us only of the Ads migration takes a while and we need to maintain both locations.

Reply:
I've created https://github.com/googleapis/gapic-generator-python/issues/1994 to follow up on whether we want to try to remove the ads templates (working with the Ads team), or refactor code so it can be shared between both template directories. Let's discuss with the ads team to understand the gaps in the standard templates before spending time on refactoring code. I've filed https://github.com/googleapis/gapic-generator-python/issues/1994 to follow up on this.